src/utils/cpUtils.ts

```diff
@@ -4,12 +4,14 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as cp from 'child_process';
+import * as os from 'os';
 import * as vscode from 'vscode';
 import { localize } from '../localize';
 
 export namespace cpUtils {
-    export async function executeCommand(outputChannel: vscode.OutputChannel | undefined, workingDirectory: string, command: string, ...args: string[]): Promise<string> {
+    export async function executeCommand(outputChannel: vscode.OutputChannel | undefined, workingDirectory: string | undefined, command: string, ...args: string[]): Promise<string> {
         let result: string = '';
+        workingDirectory = workingDirectory !== undefined ? workingDirectory : os.tmpdir();
```

Comment:
suggest: workingDirectory = workingDirectory || os.tmpdir();

Reply:
I've been avoiding that pattern because of a tslint rule. It sometimes complains with this:
```
ERROR: (strict-boolean-expressions): This type is not allowed in the operand for the '||' operator because it is always truthy. Allowed types are boolean, undefined-union, or string.
```

I definitely like the rule because it is handy in situations like this:
```typescript
enum ExampleEnum {
    a = 0,
    b = 1,
    c = 2
}

const example: ExampleEnum | undefined = tryGetEnumValue();
if (example) { // <-- tslint will complain about this. It should be "example === undefined"
    console.log('Enum is defined'); // <-- this won't display if tryGetEnumValue returns 'ExampleEnum.a'
}
```

However, I'm less sure what to do for cases like this file. For example, what should happen if workingDirectory gets passed in as an empty string? On one hand - it was explicitly passed in so why would we ignore that value? On the other hand...I'm not sure it makes sense to ever use a value of an empty string.

Like I said - I want to keep the rule on. However, I don't have a strong opinion for how we handle this specific case. Thoughts?